### PR TITLE
First version of tests & TSLint/typo fixes

### PR DESCRIPTION
**Pull request by Juuldamen**

### CODE REVIEW

**Review by quintaartsen — APPROVED** [review]

File: src/__tests__/connectionService.test.ts [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+import {UnknownDestinationException} from "../exceptions/unknownDestination.exception";
+import {Message} from "../models";
+import {ConnectionService} from "../services/concretes/connection.service";
+import {TestBodies, TestGuids, TestIps} from "./testAddress";
+
+let connectionService: ConnectionService;
+let testPort: string;
+
+beforeEach(async () => {
+    testPort = "65535";
```
Comment (RikvanSpreuwel) [inline src/__tests__/connectionService.test.ts]: Define a constant for this variable.

File: src/__tests__/message.model.test.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+let type: string;
+let originalSenderGuid: string;
+let body: string;
+let correlationId: string;
+let message: Message;
+let message2: Message;
+
+beforeAll(() => {
+    type = "ping";
```
Comment (RikvanSpreuwel) [inline src/__tests__/message.model.test.ts]: If this variable doesn't change, please consider making it a constant.

File: src/__tests__/message.model.test.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    originalSenderGuid = Guid.create().toString();
+    body = "body";
```
Comment (RikvanSpreuwel) [inline src/__tests__/message.model.test.ts]: If this variable doesn't change, please consider making it a constant.

Example (outside of the beforeAll() method):
const body = "ping";

File: src/__tests__/routingTable.model.test.ts [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+
+describe("Adding/removing a peer", () => {
+    it("Should add/remove the peer to/from the peers map", () => {
+        const peerGuid = Guid.create().toString();
+        const ip = TestIps.TEST_1;
+
+        routingTable.addPeer(peerGuid, ip, PeerType.VALIDATOR);
+        const peerNode = routingTable.peers.get(peerGuid);
+
+        expect(peerNode).not.toBeUndefined();
```
Comment (RikvanSpreuwel) [inline src/__tests__/routingTable.model.test.ts]: Please change "expect(peerNode).not.toBeUndefined();" to:

expect(peerNode).toBeDefined();

File: src/__tests__/routingTable.model.test.ts [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+
+        peersMap.set(peerGuid, new PeerNode(ip, PeerType.VALIDATOR));
+        routingTable.addPeer(peerGuid2, ip2, PeerType.VALIDATOR);
+
+        routingTable.mergeRoutingTables(peersMap);
+
+        const peerNode = routingTable.peers.get(peerGuid);
+        const peerNode2 = routingTable.peers.get(peerGuid2);
+
+        expect(peerNode).not.toBeUndefined();
```
Comment (RikvanSpreuwel) [inline src/__tests__/routingTable.model.test.ts]: Please change not.toBeUndefined() to .toBeDefined();

File: src/__tests__/routingTable.model.test.ts [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+        expect(peerNode2).not.toBeUndefined();
```
Comment (RikvanSpreuwel) [inline src/__tests__/routingTable.model.test.ts]: Please change not.toBeUndefined() to .toBeDefined();

File: src/__tests__/routingTable.model.test.ts [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+});
+
+describe("Get Peer of Type from routing table", () => {
+    it("Should return a peer from the given type", () => {
+        const peerGuid = Guid.create().toString();
+        const ip = TestIps.TEST_1;
+        routingTable.addPeer(peerGuid, ip, PeerType.VALIDATOR);
+
+        const valPeerNode = routingTable.getPeerOfType(PeerType.VALIDATOR);
+        expect(valPeerNode).not.toBeUndefined();
```
Comment (RikvanSpreuwel) [inline src/__tests__/routingTable.model.test.ts]: Please change not.toBeUndefined() to .toBeDefined();

**Review by RikvanSpreuwel — CHANGES_REQUESTED** [review]
Overall, change all .not.toBeUndefined() calls to .toBeDefined(), this does exactly the same, but removes the useless .not call.

**Review by RikvanSpreuwel — APPROVED** [review]

**Pull request merged**